src/compiler/tracing.ts

```diff
@@ -26,6 +34,8 @@ namespace ts.tracing {
             return;
         }
 
+        mode = tracingMode;

```

Comment:
Maybe if there's some `Unset` thing, assert here that it's in that state, or that it was used with `Build` and the new value is the `Build` too, etc?

(Also, due to the same confusion: if it shouldn't consume a `Project` then should it be an error with that?)

Reply:
As the code exists now, the value will never be consumed, so I'm reluctant to add a bunch of runtime checks to validate something that is observable true.  However, if you think it's important, I can make it `| undefined` and use `!` at the only read to make it clear that I've thought about what values can reach that point.

Reply:
Dropped initializer to clarify.